File: backend/application-management/src/main/java/com/cybersoft/application_management/service/FormTypeService.java
package com.cybersoft.application_management.service;

import java.util.List;
import java.util.UUID;

import com.cybersoft.application_management.dto.request.CreateFormTypeRequest;
import com.cybersoft.application_management.dto.request.UpdateFormTypeRequest;
import com.cybersoft.application_management.dto.response.FormTypeResponse;

public interface FormTypeService {
    List<FormTypeResponse> getAll();

    FormTypeResponse getById(UUID formTypeId);

    FormTypeResponse create(CreateFormTypeRequest request);

    FormTypeResponse update(UUID formTypeId, UpdateFormTypeRequest request);

    void activate(UUID formTypeId);

    void deactivate(UUID formTypeId);
}

File: frontend/src/features/form-types/api/form-type-service.ts
import type {
  CreateFormTypeRequest,
  FormTypeResponse,
  UpdateFormTypeRequest,
} from "../../../api/generated";

import formTypeApi from "./form-type-api";

export async function getFormTypes(): Promise<FormTypeResponse[]> {
  const response = await formTypeApi.getAll();

  return response.data;
}

export async function getFormTypeById(
  formTypeId: string,
): Promise<FormTypeResponse> {
  const response =
    await formTypeApi.getById(formTypeId);

  return response.data;
}

export async function createFormType(
  request: CreateFormTypeRequest,
): Promise<FormTypeResponse> {
  const response =
    await formTypeApi.create(request);

  return response.data;
}

export async function updateFormType(
  formTypeId: string,
  request: UpdateFormTypeRequest,
): Promise<FormTypeResponse> {
  const response =
    await formTypeApi.update(
      formTypeId,
      request,
    );

  return response.data;
}

export async function activateFormType(
  formTypeId: string,
): Promise<void> {
  await formTypeApi.activate(formTypeId);
}

export async function deactivateFormType(
  formTypeId: string,
): Promise<void> {
  await formTypeApi.deactivate(formTypeId);
}